File: movies/src/main/java/com/puntored/movies/entity/FilmCategory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.puntored.movies.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

/**
 *
 * @author heiderarellano
 */
@Entity
@Data
@Schema(name = "FilmCategory", description = "Relación entre una película y su género")
@Table(name = "film_category")
public class FilmCategory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Schema(description = "Identificador de la relación", example = "1")
    @Column(name = "filmcategory_id")
    private long filmcaterogyId;

    @ManyToOne
    @JoinColumn(name = "film_id")
    private Film film;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: movies-python/models/filmcategory.py
from sqlalchemy import Table, Column, ForeignKey
from sqlalchemy.sql.sqltypes import Integer
from config.database import meta, engine

filmcategory = Table(
    "filmcategory",
    meta,
    Column("filmcategory_id", Integer, primary_key=True),
    Column("film_id", Integer, ForeignKey("film.film_id")),
    Column("category_id", Integer, ForeignKey("category.category_id"))
)

meta.create_all(engine)